### DISCUSSION TITLE
Problem with upgrading git extension package from v3.x

### DISCUSSION THREAD

**Discussion by AdmiringWorm** [chocolatey-community/chocolatey-packages#2091] [Announcements]
Recently we became aware of a problem when upgrading the package gitextension when version 3.5.4 is already installed (other versions are affected as well).
There is currently not a consensus of whether we should consider this a problem in the package itself, or if this is the responsibility of the installer.

For the time being it is not being fixed in the package, but that may change after the holidays.
The current workaround if you are seeing problems when upgrading the package is to first uninstall git extensions (`choco uninstall gitextensions`), or if the package is not installed to remove it using normal manual approach.

You should then be able to upgrade/install the gitextensions package again.

Thank you to @rgl for reporting this issue.

**Comment by corbob** [comment]
For what it's worth, it looks like the MSI at least reports the issue being that you can't upgrade:

```
MSI (s) (40:34) [15:20:57:691]: Product: Git Extensions 4.0.1.15887 -- The installed version is not upgradeable to 4.0.1.15887.
Uninstall the previous version first.
Setup will now exit.
The installed version is not upgradeable to 4.0.1.15887.
Uninstall the previous version first.
Setup will now exit.
```

**Comment by Cyber1000** [comment]
FYI: seems to be this problem: https://github.com/gitextensions/gitextensions/issues/10383
(in short: Uninstall of 3.5.4 is required before installing 4.0. User settings are preserved.)

**Reply by Cyber1000** [reply]
Maybe a warning "Uninstall of 3.5.4 is required before installing 4.0. User settings are preserved." in the output would be sufficient when upgrading from <4.0 to =>4.0

**Reply by AdmiringWorm** [reply]
Unfortunately, we can not add warnings to any package versions that have already been submitted. We could add a warning in versions going forward, but there is no saying when or if the warning will be obsolete and no longer apply.

**Reply by Cyber1000** [reply]
I'm not aware what information you get within the script part, since I'm more or less choco-consumer only.
Maybe you can get the current-user-version in a prehook-script (in future versions) and give the warning whenever it is lower than version 4.0.

But maybe this is wasted time for a temporary problem (until everyone has updated) and a comment in https://community.chocolatey.org/packages/gitextensions would suffice (at least you can find the information in the comments section)

Thanks for answering!

**Reply by AdmiringWorm** [reply]
> Maybe you can get the current-user-version in a prehook-script (in future versions) and give the warning whenever it is lower than version 4.0.

pre-hook scripts are user defined scripts (can be installed through hook packages as well though), and not part of the package itself.
So using a pre-hook wouldn't be possible for this package itself.

> I'm not aware what information you get within the script part, since I'm more or less choco-consumer only.

I am honestly not 100% sure what information is available, but if the installer itself put something in the registry about which version is installed, then that can be used to determine if a warning is necessary.

> But maybe this is wasted time for a temporary problem (until everyone has updated) and a comment in [community.chocolatey.org/packages/gitextensions](https://community.chocolatey.org/packages/gitextensions) would suffice (at least you can find the information in the comments section)

That is kinda where I am leaning against
